File: core/src/main/java/net/torommo/logspy/ByType.java
package net.torommo.logspy;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Configures a log spy to record log events to a logger for given type.
 */
@Target(ElementType.PARAMETER)
@Retention(RetentionPolicy.RUNTIME)
public @interface ByType {
    /**
     * The type.
     *
     * @return The type.
     */
    Class<?> value();
}

File: core/src/main/kotlin/net/torommo/logspy/ByType.kt
package net.torommo.logspy

import kotlin.annotation.AnnotationRetention.RUNTIME
import kotlin.annotation.AnnotationTarget.VALUE_PARAMETER
import kotlin.reflect.KClass

/**
 * Configures a log spy to record log events to a logger for given type.
 * @param value the type
 */
@Target(VALUE_PARAMETER)
@Retention(RUNTIME)
@MustBeDocumented
annotation class ByType(val value: KClass<out Any>)
